src/lsp/server.rs

```diff
@@ -165,17 +180,16 @@ impl LSPServer {
 
         let check_result =
             check_internal(self.project_root.clone(), &self.project_config, true, true)?;
-        let diagnostics = check_result
-            .into_iter()
-            .filter_map(|e| {
-                if let Some(file_path) = e.file_path() {
-                    if uri_pathbuf == self.project_root.join(file_path) {
-                        return e.into();
-                    }
-                }
-                None
-            })
+        let check_external_result = check_external(&self.project_root, &self.project_config)?;
+
+        let check_diagnostics = self.filter_diagnostics_results(check_result, &uri_pathbuf);
+        let check_external_diagnostics =
+            self.filter_diagnostics_results(check_external_result, &uri_pathbuf);
```

Comment:
Just noticed, we should be able to make the filter function generic with a trait bound for `IntoIterator<Item = Diagnostic>`, then apply it after chaining the results together as owning iterators. But definitely not a blocker for this PR!